src/util/jinja_to_qweb.py

```diff
@@ -246,70 +288,78 @@ def upgrade_jinja_fields(
     sql_where_qweb_fields = [field + r"~ '(\$\{|%\s*(if|for))'" for field in qweb_fields]
     sql_where_fields = " OR ".join(sql_where_inline_fields + sql_where_qweb_fields)
 
-    templates_to_check[table_name] = []
+    setup_templates_to_check(cr)
     model = model_of_table(cr, table_name)
 
     cr.commit()  # ease the processing for PG
-    cr.execute(
+    chunk_cr = db_connect(cr.dbname).cursor()
+    chunk_cr.execute(
         f"""
         SELECT id, {name_field}, {sql_fields}
           FROM {table_name}
          WHERE {sql_where_fields}
         """
     )
-    for data in cr.dictfetchall():
-        _logger.info("process %s(%s) %s", table_name, data["id"], data[name_field])
-
-        # convert the fields
-        templates_converted = {}
+    while True:
+        data_chunk = chunk_cr.dictfetchmany(100)
+        if not data_chunk:
+            break
+        for data in data_chunk:
```

Comment:
I'd suggest you move this to a generator then use the generator in the main loop. This will minimize your diff and make it easier to review and keep a better line history.
```suggestion
    datas = itertools.takewhile(bool, map(ncr.dictfetchmany, itertools.repeat(300)))
    for data in datas:
```